http-server/src/server.rs

```diff
@@ -215,43 +215,43 @@ impl Server {
 						let (tx, mut rx) = mpsc::unbounded::<String>();
 						// Is this a single request or a batch (or error)?
 						let mut single = true;
-
-						// For reasons outlined [here](https://github.com/serde-rs/json/issues/497), `RawValue` can't be
-						// used with untagged enums at the moment. This means we can't use an `SingleOrBatch` untagged
-						// enum here and have to try each case individually: first the single request case, then the
-						// batch case and lastly the error. For the worst case – unparseable input – we make three calls
-						// to [`serde_json::from_slice`] which is pretty annoying.
-						// Our [issue](https://github.com/paritytech/jsonrpsee/issues/296).
-						if let Ok(req) = serde_json::from_slice::<JsonRpcRequest>(&body) {
-							// NOTE: we don't need to track connection id on HTTP, so using hardcoded 0 here.
-							methods.execute(&tx, req, 0).await;
-						} else if let Ok(_req) = serde_json::from_slice::<JsonRpcNotification<Option<&RawValue>>>(&body)
-						{
-							return Ok::<_, HyperError>(response::ok_response("".into()));
-						} else if let Ok(batch) = serde_json::from_slice::<Vec<JsonRpcRequest>>(&body) {
-							if !batch.is_empty() {
-								single = false;
-								for req in batch {
+						// TODO: Cut down on some noise. Is this a terrible idea?
+						type Notif<'a> = JsonRpcNotification<'a, Option<&'a RawValue>>;
```

Comment:
Thoughts on this? I'm doing this to avoid having this: `serde_json::from_slice::<Vec<JsonRpcNotification<Option<&RawValue>>>>(&body)` which is a mouthful...

Reply:
I like it, almost that we should re-export it from `types`, we would want it for the ws-server too then?

Reply:
Nah, right we don't care about notifications in ws-server